### ISSUE TITLE
Training Z-Image-Turbo LoRA using 12G VRAM ~ 😁

### ISSUE THREAD

**Issue by juntaosun** [Tongyi-MAI/Z-Image#36]
Training tools used ( Z-Image-Turbo ):
https://github.com/ostris/ai-toolkit

(1) Datasets Training Preparation: Image material with a maximum side size of 768.

🎉 To minimize VRAM usage, after extensive testing, training was successful with 6~10 images, yielding good LoRa results.  
>  A maximum image side resolution of 1024 might cause insufficient 12G VRAM. You can try it!  

(2) New Job Creation: Select Z-Image Turbo, and then set your model path. Follow the settings in the screenshot below.

<img width="1315" height="672" alt="Image" src="https://github.com/user-attachments/assets/f68ddc8e-b797-49cf-b60a-8f8e7a22e972" />

You need to enter your own trigger words! This is just an example !  
⚠️ Remember to set the Transformer Offload to 0%; we won't use it because it will throw an error, and we're unsure if this is a bug.  

<img width="1668" height="351" alt="Image" src="https://github.com/user-attachments/assets/3fd4cd1a-91cb-4c92-892d-3901c36385af" />

Correction: Learning Rate 0.0001 ~ 0.0002 !  
👉 After extensive testing, 2000 steps is a suitable value. 

<img width="1522" height="391" alt="Image" src="https://github.com/user-attachments/assets/570e7bf2-74f7-4611-9c7f-4fda0f256c49" />

Datasets: Cache Latents and Resolution 512, 768 ~
>  A maximum image side resolution of 1024 might cause insufficient 12G VRAM. You can try it!  

<img width="693" height="228" alt="Image" src="https://github.com/user-attachments/assets/c0870a04-8a76-45f4-9f39-ea9be49660f9" />

<img width="568" height="222" alt="Image" src="https://github.com/user-attachments/assets/7a236473-dca4-4699-a274-2f1541759862" />

👆 As you can see, training started successfully on 12G VRAM, and the speed is quite good !👇

```
first_lora_v1:  26%|##5       | 519/2000 [24:43<1:10:54,  2.87s/it, lr: 2.0e-04 loss: 3.811e-01]
```
Training speed is approximately 2~3 seconds/it, 2000 steps take about 1 - 2 hour to complete.

Finally: Wishing users with low VRAM success in training their own z-image LoRA!  
Thanks to ai-toolkit and z-image, have fun!  If you have better training settings, please share! 🤗

<img width="331" height="303" alt="Image" src="https://github.com/user-attachments/assets/b6b6c3f9-025b-46db-a4ea-9b6b96554f68" />

> This is my second z-image LoRA. ai-toolkit\output  

Now, you can use it in ComfyUI via the Lora loader !  Z-Image-Turbo ~

**Comment by hdhacker58** [comment]
I am wondering whether this can be done by purely python code (but not gui), as I am using a GPU in the server which doesn't have UI supporting.

**Comment by juntaosun** [comment]
@hdhacker58 This is a Python environment combined with Node.js. After running, it will launch a web UI, which you can then use in your browser.

**Comment by hdhacker58** [comment]
@juntaosun  Thanks but my server doesn't support for Web UI. I may try to use pure cli method for fine-tuning

**Comment by metal3d** [comment]
> [@juntaosun](https://github.com/juntaosun) Thanks but my server doesn't support for Web UI. I may try to use pure cli method for fine-tuning

I don't understand how you can connect your server with SSH and not 8675 port which is the UI port.
As soon as you launched the UI, you should be able to connect the UI.

**Comment by jdsguam** [comment]
This was incredibly useful for someone who has no idea what he's doing.  Running my first job now, got an hour or so to kill.  If this works and all I have to work on is my dataset, I'll be stoked!

**Comment by Justnormalguy** [comment]
What is this model better suited for: training a LoRA for realistic people or for anime characters?

**Comment by weathon** [comment]
If I can do it on code it will be better as I want to modify it from SFT to DPO
